File: validNumber.py
class Solution:
    # @param s, a string
    # @return a boolean
    def isNumber(self, s):
        i, n = 0, len(s)
        while(i < n and s[i] == ' '): i += 1
        if(i < n and (s[i] == '-' or s[i] == '+')): i += 1
        isNumber = False
        while(i < n and s[i].isdigit()): 
            i += 1; isNumber = True  
        if(i < n and s[i] == '.'): 
            i += 1
            while(i < n and s[i].isdigit()):
                i += 1; isNumber = True
        if(isNumber and i < n and s[i].lower() == 'e'):
            i += 1
            if(i < n and (s[i] == '-' or s[i] == '+')): i += 1
            isNumber = False 
            while(i < n and s[i].isdigit()):
                i += 1; isNumber = True
        while(i < n and s[i] == ' '): i += 1
        return isNumber and i == n

File: Array/ValidNumber.java

/** Validate if a given string is numeric. */  

class Solution(){
  
  public boolean isNumber(String s){
    int i = 0, len = s.length(); 
    boolean output = false; 
    
    while(i < len && s.charAt(i) == ' ') i ++ ; 
    
    if(i < len && (s.charAt(i) == '+' || s.charAt(i) == '-' )) i ++; 
    
    while(i < len && Character.isDigit(s.charAt(i))){
      i ++; output = true; 
    } 
    
    if(i < len && s.charAt(i) == '.'){
      i++; 
      while(i < len && Character.isDigit(s.charAt(i))){
        i ++; output = true; 
      } 
    }
    
    if(output && i < len && (s.charAt(i) == 'e' || s.charAt(i) == 'E')){
      i ++; output = false; 
      if(i < len && (s.charAt(i) == '+' || s.charAt(i) == '-' )) i ++; 
      while(i < len && Character.isDigit(s.charAt(i))){
        i ++; output = true; 
      } 
    }
    
    while(i < len && s.charAt(i) == ' ') i ++; 
    return output && i == len; 
  }
  
	public static void main(String[] args) {
	  /** test cases */ 
		String[][] testCase = {{"0", " 0.1 ", " 2e10"}, {"1 a", "abc", " "}};
		for(int i = 0; i < testCase.length; i++ ){
			for(String s : testCase[i]){
				System.out.print(new Solution().isNumber(s) + " ");
			}			
			System.out.println();
		}
	}
}
  
  
